pilot/cmd/pilot-agent/status/server.go

```diff
@@ -294,7 +294,18 @@ func (s *Server) handlePprofTrace(w http.ResponseWriter, r *http.Request) {
 	pprof.Trace(w, r)
 }
 
+const (
+	ReadinessTimeout = time.Second * 3 // Default Readiness probe timeout. It is set the same in helm charts.
+)
+
 func (s *Server) handleReadyProbe(w http.ResponseWriter, _ *http.Request) {
+	t0 := time.Now()
+	defer func() {
+		if time.Since(t0) > ReadinessTimeout {
+			log.Warnf("handleReadyProbe timed out")
```

Comment:
A bit confused, don't we already log the timeout in `Envoy proxy is NOT ready` line?

Reply:
1s + 3s could exceed 3s, in this case  without any error log

Reply:
so the `err := s.ready.Check()` could take more than 3s but succedded in the end. However, on the kubelet side, it is already timed out